tests/libraries/token/test_stake_token.py

```diff
@@ -0,0 +1,325 @@
+    assert stakeToken.buy(designatedReporterStake)
+    assert stakeToken.balanceOf(testerAddress) == newTesterBalance
+
+def test_stake_token_redeem(tokenFixture, universe, cash):
+    numTicks = 10 ** 18
+    denominationToken = cash
+    feePerEthInWei = 10**10
+    endTime = long(tokenFixture.chain.head_state.timestamp + timedelta(days=1).total_seconds())
+
+    market_other = tokenFixture.createBinaryMarket(universe, endTime, feePerEthInWei, denominationToken, tester.a8, numTicks, tester.k9)
+    market_forking = tokenFixture.createBinaryMarket(universe, endTime, feePerEthInWei, denominationToken, tester.a1, numTicks)
+    designatedReporterStake = universe.getDesignatedReportStake()
+    reputationToken = tokenFixture.applySignature('ReputationToken', universe.getReputationToken())
+    stakeToken_other = tokenFixture.getStakeToken(market_other, [0, numTicks])
+    original_dr_other = stakeToken_other.balanceOf(tester.a8) 
+    original_market_creator_other = stakeToken_other.balanceOf(tester.a9) 
+
+    # Designated Report votes that Market is invalid
+    proceedToDesignatedReporting(tokenFixture, universe, market_other, [0, numTicks])
+    assert market_other.getReportingState() == tokenFixture.contracts['Constants'].DESIGNATED_REPORTING()
+    stakeToken_other.buy(designatedReporterStake, sender=tester.k8)
+    assert stakeToken_other.balanceOf(tester.a8) == designatedReporterStake
+    assert market_other.isContainerForStakeToken(stakeToken_other.address)
+
+    # kick this market into the DESIGNATED_REPORTING phase
+    proceedToDesignatedReporting(tokenFixture, universe, market_forking, [0, numTicks])
+
+    proceedToRound1Reporting(tokenFixture, universe, market_forking, False, tester.a6, [0, numTicks], [numTicks, 0])
+    assert market_forking.getReportingState() == tokenFixture.contracts['Constants'].ROUND1_REPORTING()
+
+    proceedToRound2Reporting(tokenFixture, universe, market_forking, False, tester.a6, tester.k3, [numTicks, 0], [0, numTicks], tester.k2, [0, numTicks], [numTicks, 0])
+    assert market_forking.getReportingState() == tokenFixture.contracts['Constants'].ROUND2_REPORTING()
+
+    # Proceed to the FORKING phase
+    proceedToForking(tokenFixture, universe, market_forking, True, tester.k1, tester.k3, tester.k4, [0,numTicks], [numTicks,0], tester.k2, [numTicks,0], [numTicks,0], [0,numTicks])
+    assert market_forking.getReportingState() == tokenFixture.contracts['Constants'].FORKING()
+
+    # during a fork non-forking markets should be in limbo and allow users to redeemDisavowedTokens
+    assert market_other.getReportingState() == tokenFixture.contracts['Constants'].AWAITING_FORK_MIGRATION()
+    stakeToken_non_forking = tokenFixture.applySignature('StakeToken', market_other.getStakeToken([0, numTicks]))
+    assert market_other.isContainerForStakeToken(stakeToken_non_forking.address) == True
+
+    # // :TODO Finish up test when redeem has been fixed
```

Comment:
@nuevoalex Is redeem fully working now?

Reply:
I'm working on finishing it up now. That said I'm also writing tests that probably cover what the TODO is hoping to, so I'd say this TODO could just be removed :)